core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -311,7 +311,24 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
       getV2WorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.read, Option(attrSpecs)) flatMap {
         workspaceContext =>
           dataSource.inTransaction { dataAccess =>
-            val wsmContext = wsmService.optimizedFetchAggregatedWorkspace(workspaceContext, ctx)
+            // some GCP workspaces, like those with linked snapshots, have a stub WSM workspace
+            val wsmContext =
+              try
+                wsmService.fetchAggregatedWorkspace(workspaceContext, ctx)
+              catch {
+                case e: AggregateWorkspaceNotFoundException =>
+                  // return workspace with no WSM information for gcp workspace
+                  if (workspaceContext.workspaceType == WorkspaceType.RawlsWorkspace) {
+                    AggregatedWorkspace(workspaceContext,
+                                        Some(workspaceContext.googleProjectId),
+                                        azureCloudContext = None,
+                                        policies = List.empty
+                    )
+                  } else {
+                    // bubble up an MC workspace exception
+                    throw e
+                  }
+              }
```

Comment:
This code is correct and in good shape, and I would not block the PR on changes here. However, the use of `try/catch` is Java-idiomatic, where the use of `Try/Success/Failure` is Scala-idiomatic. To be a good Scala citizen, you'd have something like this:

```suggestion
val wsmContext =
              Try(wsmService.fetchAggregatedWorkspace(workspaceContext, ctx)) match {
                case Success(found) => found
                case Failure(notFound: AggregateWorkspaceNotFoundException)
                    if workspaceContext.workspaceType == WorkspaceType.RawlsWorkspace =>
                  // return workspace with no WSM information for gcp workspace
                  AggregatedWorkspace(workspaceContext,
                                      Some(workspaceContext.googleProjectId),
                                      azureCloudContext = None,
                                      policies = List.empty
                  )
                case Failure(x) => throw x // bubble up an MC workspace exception
              }
```